### PR TITLE
Add asserts to entry and setEntry as a out of bound without an exception would be possible.

### PR DESCRIPTION
**Pull request by Fox32**
Add asserts to entry and setEntry as a out of bound without an exception would be possible.

Ironically there is a test that is going out of bounds but didn't failed previously and still works fine after fixing the bounds...

### CODE REVIEW

File: lib/src/vector_math/matrix2.dart [hunk 1]
```diff
@@ -48,10 +48,18 @@ class Matrix2 {
   int index(int row, int col) => (col * 2) + row;
 
   /// Value at [row], [col].
-  double entry(int row, int col) => storage[index(row, col)];
+  double entry(int row, int col) {
+    assert(row >= 0 && row < dimension);
```
Comment (johnmccutchan) [inline lib/src/vector_math/matrix2.dart]: please rewrite like this: ((row >= 0) && (row < dimension))

Comment (johnmccutchan) [inline lib/src/vector_math/matrix2.dart]: (here and elsewhere)

**Comment by johnmccutchan** [comment]
LGTMwC

**Comment by Fox32** [comment]
@johnmccutchan  Done

**Pull request merged**